### PR TITLE
Prefer Scala Numeric classes over Java classes

### PR DESCRIPTION
**Pull request by gkleiman**
Now that we have play-json, I see no reason for using java.lang.integer and
java.lang.double instead of Scala's classes.

cc/ @aquamatthias

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/EndpointsHelper.scala [hunk 1]
```diff
@@ -35,7 +35,7 @@ object EndpointsHelper {
           sb.append(cleanId).append(delimiter).append(port).append(delimiter)
 
           for (task <- tasks if task.getStatus.getState == TaskState.TASK_RUNNING) {
-            val taskPort = Option(task.getPortsList.get(i): java.lang.Integer).getOrElse(java.lang.Integer.valueOf(0))
+            val taskPort: Integer = Option(task.getPortsList.get(i): Integer).getOrElse(0)
             sb.append(task.getHost).append(':').append(taskPort).append(delimiter)
```
Comment (kolloch) [inline src/main/scala/mesosphere/marathon/api/EndpointsHelper.scala]: `Integer` is `java.lang.Integer`... `Int` is Scala.

File: src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala [hunk 2]
```diff
@@ -147,9 +145,9 @@ trait Formats
     Reads.of[Int].map(Int.box),
     Writes[Integer] { i => JsNumber(i.intValue) }
   )
-  implicit lazy val DoubleFormat: Format[JDouble] = Format(
+  implicit lazy val DoubleFormat: Format[java.lang.Double] = Format(
```
Comment (kolloch) [inline src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala]: Unneeded, IntegerFormat as well.

**Comment by kolloch** [comment]
Scalas integer type is `Int`

**Comment by gkleiman** [comment]
@kolloch thanks for the review :blush:

I addressed your comments, now we have `Int` almost everywhere =).

**Comment by kolloch** [comment]
Thank you! Nice endeavour! :+1: :shipit:

**Pull request merged**